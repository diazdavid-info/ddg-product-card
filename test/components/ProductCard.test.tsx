import React from 'react'
import renderer from 'react-test-renderer'
import { ProductCard } from '../../src'
// @ts-ignore
import { product } from '../data/products'

const { act } = renderer

describe('ProductCard', () => {
    test('Should show correctly', () => {
        const wrapper = renderer.create(
            <ProductCard product={ product }>
                {
                    () => (
                        <h1>Product card</h1>
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot()
    })

    test('Should increment counter', () => {
        const wrapper = renderer.create(
            <ProductCard product={ product }>
                {
                    ({ count, increaseBy }) => (
                        <>
                            <h1>Product Card</h1>
                            <span>{ count }</span>
                            <button onClick={ () => increaseBy(1) }>test</button>
                        </>
                    )
                }
            </ProductCard>
        )

        let tree: any = wrapper.toJSON()
        expect(tree).toMatchSnapshot()

        act(() => {
            tree.children[2].props.onClick()
        })

        tree = wrapper.toJSON()

        expect(tree.children[1].children[0]).toBe('1')
    })
})
