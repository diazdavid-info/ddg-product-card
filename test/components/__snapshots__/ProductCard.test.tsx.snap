// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Should increment counter 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    test
  </button>
</div>
`;

exports[`ProductCard Should show correctly 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product card
  </h1>
</div>
`;
